### PR TITLE
Updates for pythia8 

### PR DESCRIPTION
**Pull request by iarspider**
* Changed to new URL
* Added Version 8.306
* Added `setup_dependent_build_environment` method

### CODE REVIEW

**Comment by iarspider** [comment]
@spackbot run pipeline

**Review by alalazo — CHANGES_REQUESTED** [review]
Looking at the homepage, it seems the versions are 8.306 and not 8306?

**Comment by iarspider** [comment]
> 
> 
> Looking at the homepage, it seems the versions are 8.306 and not 8306?

Yes. I decided not to change it since the recipe has been around for some time... I can change the versions to follow the website, if you think that's ok.

**Comment by alalazo** [comment]
@ChristianTackeGSI Do you have any preference on that?

**Comment by ChristianTackeGSI** [comment]
Thanks for pinging me! :-)

No, I don't have any preferences, really.

Just need to change over our local environments. Nothing real big.

File: var/spack/repos/builtin/packages/pythia8/package.py [hunk 1]
```diff
@@ -71,6 +72,16 @@ def configure_args(self):
...
+
+        return url + '/' + dirname + '/' + fname
+
     def setup_dependent_run_environment(self, env, dependent_spec):
         env.set('PYTHIA8', self.prefix)
         env.set('PYTHIA8DATA', self.prefix.share.Pythia8.xmldoc)
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.set('PYTHIA8', self.prefix)
+        env.set('PYTHIA8DATA', self.prefix.share.Pythia8.xmldoc)
```
Comment (ChristianTackeGSI) [inline var/spack/repos/builtin/packages/pythia8/package.py]: Just an idea:

Could we either have a `setup_common_env` (and call it from the various `setup_*` to not repeat all this setup code?

Or maybe something like this (they have the same signature, so this should work):
```suggestion

    setup_dependent_build_environment = setup_dependent_run_environment
```

Comment (iarspider) [reply var/spack/repos/builtin/packages/pythia8/package.py]: Sure.

**Review by alalazo — APPROVED** [review]

**Pull request merged**